pkg/azurecluster/validate_update.go

```diff
@@ -67,18 +69,31 @@ func (a *UpdateValidator) Validate(ctx context.Context, request *v1beta1.Admissi
 		return microerror.Mask(err)
 	}
 
+	return a.validateRelease(ctx, azureClusterOldCR, azureClusterNewCR)
+}
+
+func (a *UpdateValidator) Log(keyVals ...interface{}) {
+	a.logger.Log(keyVals...)
+}
+
+func (a *UpdateValidator) validateRelease(ctx context.Context, azureClusterOldCR *capzv1alpha3.AzureCluster, azureClusterNewCR *capzv1alpha3.AzureCluster) error {
 	oldClusterVersion, err := semverhelper.GetSemverFromLabels(azureClusterOldCR.Labels)
 	if err != nil {
-		return microerror.Maskf(errors.ParsingFailedError, "unable to parse version from AzureConfig (before edit)")
+		return microerror.Maskf(errors.ParsingFailedError, "unable to parse version from the AzureCluster being updated")
 	}
 	newClusterVersion, err := semverhelper.GetSemverFromLabels(azureClusterNewCR.Labels)
 	if err != nil {
-		return microerror.Maskf(errors.ParsingFailedError, "unable to parse version from AzureConfig (after edit)")
+		return microerror.Maskf(errors.ParsingFailedError, "unable to parse version from applied AzureCluster")
 	}
 
-	return releaseversion.Validate(ctx, a.ctrlClient, oldClusterVersion, newClusterVersion)
-}
+	if !newClusterVersion.Equals(oldClusterVersion) {
+		// Upgrade is triggered, let's check if we allow it
+		if capiconditions.IsTrue(azureClusterOldCR, aeconditions.CreatingCondition) {
+			return microerror.Maskf(errors.InvalidOperationError, "upgrade cannot be initiated now, AzureCluster condition %s is set to True, cluster is currently being created", aeconditions.CreatingCondition)
+		} else if capiconditions.IsTrue(azureClusterOldCR, aeconditions.UpgradingCondition) {
+			return microerror.Maskf(errors.InvalidOperationError, "upgrade cannot be initiated now, AzureCluster condition %s is set to True, cluster is already being upgraded", aeconditions.UpgradingCondition)
```

Comment:
AzureCluster does not have Creating and Upgrading conditions, here we should fetch Cluster CR and check its conditions.